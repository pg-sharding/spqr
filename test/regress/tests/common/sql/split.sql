CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh2 FOR DISTRIBUTION ds1;
CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
ALTER DISTRIBUTION ds1 ATTACH RELATION test DISTRIBUTION KEY id;

SPLIT KEY RANGE krid3 FROM krid1 BY 5;
SHOW key_ranges;

LOCK KEY RANGE krid1;
SPLIT KEY RANGE krid4 FROM krid1 BY 5;
UNLOCK KEY RANGE krid1;

SPLIT KEY RANGE krid4 FROM krid1 BY 50;

SPLIT KEY RANGE krid4 FROM krid2 BY 5;

DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
