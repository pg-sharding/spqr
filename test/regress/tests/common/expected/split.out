CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE krid1 FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 0
(1 row)

CREATE KEY RANGE krid2 FROM 11 ROUTE TO sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 11
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION test DISTRIBUTION KEY id;
      attach table       
-------------------------
 relation name   -> test
 distribution id -> ds1
(2 rows)

SPLIT KEY RANGE krid3 FROM krid1 BY 5;
    split key range    
-----------------------
 key range id -> krid1
 bound        ->      +
 
(2 rows)

SHOW key_ranges;
 Key range ID | Shard ID | Distribution ID | Lower bound 
--------------+----------+-----------------+-------------
 krid1        | sh1      | ds1             | 0
 krid2        | sh2      | ds1             | 11
 krid3        | sh1      | ds1             | 5
(3 rows)

LOCK KEY RANGE krid1;
    lock key range     
-----------------------
 key range id -> krid1
(1 row)

SPLIT KEY RANGE krid4 FROM krid1 BY 5;
ERROR:  key range "krid1" is locked.
UNLOCK KEY RANGE krid1;
   unlock key range    
-----------------------
 key range id -> krid1
(1 row)

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

